### PR TITLE
8314830: runtime/ErrorHandling/ tests ignore external VM flags

### PR DESCRIPTION
**Pull request by mseledts**
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314830](https://bugs.openjdk.org/browse/JDK-8314830): runtime/ErrorHandling/ tests ignore external VM flags (**Sub-task** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15767/head:pull/15767` \
`$ git checkout pull/15767`

Update a local copy of the PR: \
`$ git checkout pull/15767` \
`$ git pull https://git.openjdk.org/jdk.git pull/15767/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15767`

View PR using the GUI difftool: \
`$ git pr show -t 15767`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15767.diff">https://git.openjdk.org/jdk/pull/15767.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15767#issuecomment-1721675430)

### CODE REVIEW

**Comment by mseledts** [comment]
Please review this change that adds @requires vm.flagless to most of the tests under test/hotspot/jtreg/runtime/ErrorHandling/.
The goal of these tests is to create a specific error condition and then to check the correctness of the error handling message. Each test uses carefully crafted command line to force specific error condition. Such tests do not intend to pass test JDK/JVM flags to child VM under test, therefore they should be marked with vm.flagless.

VeryEarlyAssertTest.java seems to be the only exception in this case - I have updated it to use ProcessTools method that passes JVM flags to child VM under test.

**Comment by mseledts** [comment]
@lmesnik Please take a look when you have a chance.

File: test/hotspot/jtreg/runtime/ErrorHandling/VeryEarlyAssertTest.java [hunk 1]
```diff
@@ -49,8 +49,7 @@ public class VeryEarlyAssertTest {
   public static void main(String[] args) throws Exception {
 
 
-    ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
-            "-version");
+    ProcessBuilder pb = ProcessTools.createTestJvm("-version");
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/ErrorHandling/VeryEarlyAssertTest.java:52]: I don't think this is necessary. IIUC this test triggers an assertion failure during the static initialization that occurs in debug.o which will happen when libjvm is loaded, well before any argument processing could occur, or have any impact.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
I agree with the reasoning here, except for VeryEarlyAssert.

Thanks.

File: test/hotspot/jtreg/runtime/ErrorHandling/VeryEarlyAssertTest.java (hunk 1 above)
Comment (mseledts) [reply test/hotspot/jtreg/runtime/ErrorHandling/VeryEarlyAssertTest.java]: David, thank you for providing the details. I agree, and have updated the test.

**Review by dholmes-ora — APPROVED** [review]
Looks good. Thanks

**Review by lmesnik — APPROVED** [review]

**Comment by mseledts** [comment]
David, Leonid,
Thank you for review.

/integrate

**Pull request closed without merging**